docs/reference/spar-braindump.md

```diff
@@ -0,0 +1,170 @@
+work fine.
+
+
+### common misconceptions
+
+
+#### an email address can be one of two things
+
+the email used for saml auth is only a name, and never used for
+sending out emails, and does not show as the email address of the user
+in the team settings.
+
+RATIONALE: emails that are passed in from an external identity
+provider must be trusted, so the user cannot have them as an actual
+email address that wire is sending emails to.
+
+POSSIBLE FEATURE: we could authenticate the emails sent in from the
+identity provider in the same way we are doing that for
+password-authenticated non-team users: email receives a link
+containing a crypto token, user clicks on link if the email is
+authentic, email gets authenticated.
+
+
+#### scim, provisioning, metadata
+
+for changing the user information (name, handle, email, ...), saml
+isn't enough.  the identity management software (AD in this case, or
+some add-on) needs to support scim.  we *could* support doing that via
+saml, but the part of the standards that are needed for that are even
+in worse shape than the ones for the authentication bits, and it would
+not lead to a good user experience.  so instead we require users to
+adopt the more robust and contemporary scim standard.
+
+
+#### we don't support binding password/phone-auth'ed users to saml yet
+
+to keep track of whether we have, see https://github.com/zinfra/backend-issues/issues/731
+
+
+
+## application logic
+
+### deleting users that exist on spar
+
+For scim- or saml-created users, there are three locations for user data:
+
+- `brig.user` (and a few things associated with that on brig and galley)
+- `spar.user`
+- `spar.scim_user`
+
+The single source of truth is `brig.user`.  Dangling entries in the
+other places are allowed, and must be checked by the application code
+for danglingness.  ([test case for
+scim](https://github.com/wireapp/wire-server/blob/010ca7e460d13160b465de24dd3982a397f94c16/services/spar/test-integration/Test/Spar/Scim/UserSpec.hs#L239-L308);
+[test case for
+saml](https://github.com/wireapp/wire-server/blob/293518655d7bae60fbcb0c4aaa06034785bfb6fc/services/spar/test-integration/Test/Spar/APISpec.hs#L742-L795))
+
+For the semantics of interesting corner cases, consult [the test
+suite](https://github.com/wireapp/wire-server/blob/develop/services/spar/test-integration/Test/Spar/APISpec.hs).
+If you can't find what you're looking for there, please add at least a
+pending test case explaining what's missing.
+
+Side note: Users in brig carry an enum type
+[`ManagedBy`](https://github.com/wireapp/wire-server/blob/010ca7e460d13160b465de24dd3982a397f94c16/libs/brig-types/src/Brig/Types/Common.hs#L393-L413);
+see also {#DevScimOneWaySync}.  This is a half-implemented feature for
+managing conflicts between changes via scim vs. changes from wire
+clients; and does currently not affect deletability of users.  (grep
+`wire-server` for up-to-date information.)
```

Comment:
What do you mean here with up to date information?

Reply:
i mean "it is conceivable that this documentation is out of date, please consult the source".

i'm emberrassed now.  remove that bracket?